cluster/sbin/collect-debug-artifacts.sh

```diff
@@ -134,41 +170,44 @@ function collect_on_remote {
   collector_host="$5"
   collector_dir="$6"
   verbose="$7"
+  start_epoch="$8"
+  end_epoch="$9"
 
-  tmp_dir="$(mktemp -d --tmpdir="`pwd`" data.XXXX)"
+  tmp_dir="$(mktemp -d --tmpdir="$data_dir" data.XXXX)"
   retval=$?
   if [ ! -d ${tmp_dir} ]; then
     echo "FAILED TO CREATE tmp dir on ${host} at ${data_dir} with errno ${retval}"
+    return
```

Comment:
The better way to break out would be "exit 1" (or some other non-zero code)
Otherwise function will simply return and caller will continue execution thereon.